app/core/SDKConnect/SDKConnect.ts

```diff
@@ -413,10 +424,16 @@ export class Connection extends EventEmitter2 {
         // It will wait until user accept/reject the connection request.
         try {
           await this.checkPermissions(message);
-          while (!this.isReady) {
-            await wait(1000);
+          if (!this.receivedDisconnect) {
+            while (!this.isReady) {
+              await wait(1000);
+            }
+            this.sendAuthorized();
+          } else {
+            // Reset state to continue communication after reconnection.
+            this.isReady = true;
+            this.receivedDisconnect = false;
```

Comment:
```suggestion
  if (!this.receivedDisconnect) {
    let attempts = 0;

    while (!this.isReady && attempts < 10) {
      await wait(1000);
      attempts++;
    }

    this.sendAuthorized();
  } else {
    // Reset state to continue communication after reconnection.
    this.isReady = true;
    this.receivedDisconnect = false;
  }
```

Are we sure that we don't want to create a max attempts logic?

LMk WDYT? :)